src/Npgsql/TypeMapping/BuiltInTypeHandlerResolver.cs

```diff
@@ -344,7 +343,7 @@ internal BuiltInTypeHandlerResolver(NpgsqlConnector connector)
             "pg_lsn"     => PgLsnHandler(),
             "tid"        => TidHandler(),
             "char"       => InternalCharHandler(),
-            "record"     => RecordHandler(),
+            "record"     => new UnsupportedHandler(PgType("record"), $"Records aren't supported; please call {nameof(NpgsqlSlimDataSourceBuilder.EnableRecord)} on {nameof(NpgsqlSlimDataSourceBuilder)} to enable records."),
```

Comment:
Can cache this like the other handlers, though it's true that it's unlikely any program will actually "work" with this path.

Reply:
Don't think there is any point: we're already going to cache it at `TypeMapper`, and technically that's going to be a bug in user code, so...